streaming/base/converters/dataframeToMDS.py

```diff
@@ -0,0 +1,250 @@
+            Should be in the range (0, 1). Default is -1.0 (no sampling).
+        columns (Dict[str, str]): Sample columns.
+        keep_local (bool): If the dataset is uploaded, whether to keep the local dataset directory
+            or remove it after uploading. Defaults to ``False``.
+        compression (str, optional): Optional compression or compression:level. Defaults to
+            ``None``.
+        hashes (List[str], optional): Optional list of hash algorithms to apply to shard files.
+            Defaults to ``None``.
+        size_limit (Union[int, str], optional): Optional shard size limit, after which point to start a new
+            shard. If ``None``, puts everything in one shard. Can specify bytes
+            human-readable format as well, for example ``"100kb"`` for 100 kilobyte
+            (100*1024) and so on. Defaults to ``1 << 26``
+        udf_iterable (Callable or None): A user-defined function that returns an iterable over the dataframe. ppfn_kwargs is the k-v args for the method. Default is None.
+        udf_kwargs (Dict): Additional keyword arguments to pass to the pandas processing
+            function if provided. Default is an empty dictionary.
+
+    Returns:
+        None
+
+    Raises:
+        ValueError: If dataframe is not provided
+
+    Note:
+        - The method creates a SparkSession if not already available.
+        - If 'sample_ratio' is provided, the input data will be randomly sampled.
+        - The 'ppfn_kwargs' dictionaries can be used to pass additional
+          keyword arguments to the udf_iterable.
+    """
+
+    def write_mds(iterator: Iterable):
+
+        id = TaskContext.get().taskAttemptId()  # pyright: ignore
+        if isinstance(mds_path, str):  # local
+            output = os.path.join(mds_path, f'{id}')
+            out_file_path = output
+        else:
+            output = (os.path.join(mds_path[0], f'{id}'), os.path.join(mds_path[1], f'{id}'))
+            out_file_path = output[0]
+
+        mds_kwargs = {
+            'out': output,
+            'columns': columns,
+            'keep_local': True,
+            'compression': compression,
+            'hashes': hashes,
+            'size_limit': size_limit
+        }
+        for k, v in mds_kwargs.items():
+            if v is None:
+                del mds_kwargs[k]
+
+        with MDSWriter(**mds_kwargs) as mds_writer:
+            for pdf in iterator:
+                if udf_iterable is not None:
+                    d = udf_iterable(pdf, **udf_kwargs)  # pyright: ignore
+                else:
+                    d = pdf.to_dict('records')
+                assert is_iterable(
+                    d), f'pandas_processing_fn needs to return an iterable instead of a {type(d)}'
+
+                for row in d:
+                    mds_writer.write(row)
+        yield pd.DataFrame(pd.Series([out_file_path], name='mds_path'))
+
+    if not dataframe:
+        raise ValueError(f'input dataframe is none!')
```

Comment:
This mean's it is false, not that it is None. The type should be printed out too.